core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -336,21 +369,23 @@ func (ec *EthConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
 
 	ec.lggr.Debugw(fmt.Sprintf("Fetching receipts for %v transaction attempts", len(attempts)), "blockNum", blockNum)
 
-	attemptsByAddress := make(map[gethCommon.Address][]EthTxAttempt)
+	attemptsByAddress := make(map[string][]EthTxAttempt[ADDR, TX_HASH])
 	for _, att := range attempts {
-		attemptsByAddress[att.EthTx.FromAddress] = append(attemptsByAddress[att.EthTx.FromAddress], att)
+		attemptsByAddress[att.EthTx.FromAddress.String()] = append(attemptsByAddress[att.EthTx.FromAddress.String()], att)
 	}
 
 	for from, attempts := range attemptsByAddress {
-		minedTransactionCount, err := ec.getMinedTransactionCount(ctx, from)
+		// Get fromAddress from first attempt
+		fromAddr := attempts[0].EthTx.FromAddress
+		minedTransactionCount, err := ec.getMinedTransactionCount(ctx, fromAddr)
```

Comment:
It might be nice to have a method for getting a `comparable` key, rather than stringifying everything. Then underlying types which are comparable can just return themselves, and other types can String() or whatever is appropriate for their case.